pkg/catalogv2/content/content.go

```diff
@@ -179,6 +186,20 @@ func (c *Manager) Icon(namespace, name, chartName, version string) (io.ReadClose
 		return nil, "", err
 	}
 
+	// Check if the repositories are managed by Rancher and if the system catalog is bundled.
+	// If so, check if the icons are available locally.
+	// If the icons are not available locally, return an error and use the default fallback logo.
+	if settings.SystemCatalog.Get() == "bundled" {
```

Comment:
@rohitsakala 
According to the Rancher official documentation: https://ranchermanager.docs.rancher.com/getting-started/installation-and-upgrade/other-installation-methods/air-gapped-helm-cli-install/install-rancher-ha

![image](https://github.com/rancher/rancher/assets/127259813/0128140d-90a5-44da-8fd3-f925002f8727)

The best option we have to check is the `useBundledSystemChart -> SystemCatalog == "bundled"` which is already implemented.

The `isBundled` function you suggested: https://github.com/rancher/rancher/blob/release/v2.9/pkg/catalogv2/git/utils.go#L36

Checks for: `System Bundled Charts are system charts, rancher-charts and partner-charts that come by default when you install rancher.`

If we also add this check, third-party user-added repositories would still try to reach the internet in airgap mode. 

I suggest we don't add it.

This way in Airgap mode we will never try to fetch the internet for any Helm repositories.

Reply:
Yes, third-party user-added repositories can fetch from internet even in airgap mode. That is the ideal scenario. 

If you see the description of `userBundledSystemChart` it specifically talks about system helm charts. ie. rancher-charts, partner-charts etc. That field talks about packaging default helm charts into rancher pod so that they need not be fetched from internet.

Reply:
Ah ok changing right away